### DISCUSSION TITLE
Need Help: jars Conflict

### DISCUSSION THREAD

**Discussion by pontamilselvan** [featurehub-io/featurehub#410] [Q&A] [answered]
When i try to implement jersey client in one of existing project , getting conflict with one of existing jars. (**jaxws-rt-2.2.8-promoted-b146.jar**)

getting below error... any idea how to fix this?

May 26, 2021 6:21:55 PM com.sun.xml.ws.transport.http.servlet.WSServletDelegate <init>
INFO: WSSERVLET14: JAX-WS servlet initializing
May 26, 2021 6:21:56 PM com.sun.xml.ws.server.sei.TieHandler createResponse
SEVERE: METHOD_NOT_ALLOWED
java.lang.NoSuchFieldError: METHOD_NOT_ALLOWED
     at cd.connect.openapi.support.ApiClient.<clinit>(ApiClient.java:579)
     at io.featurehub.client.jersey.JerseyClient.<init>(JerseyClient.java:59)
     at io.featurehub.client.jersey.JerseyClient.<init>(JerseyClient.java:43)

**Comment by rvowles** [comment, marked as answer]
Could you look up the javax.ws.rs.core.Response class and find out which jar it is coming from and let me know and I hope I can help? Specifically which dependency and which version so I can grab it from Maven Central and check it out?

**Reply by pontamilselvan** [reply]
i am seeing javax.ws.rs.core.Response in below two jars. thats why conflict i guess

jersey-core-1.17.1.jar
jakarta.ws.rs-api-2.1.5.jar


My project heavily dependent on jersey-core-1.17.1.jar... and it seems featurehub needs jakarta.ws.rs-api-2.1.5.jar...

is there any solution for this?

**Reply by rvowles** [reply]
If it is just your server code that needs it, I would probably recommending copying the SDK Jersey class into your project. The APIClient library is only used for setting features using the test API endpoint, so it would be easily removed, but I haven't used Jersey 1.x for many years so I'm not really sure...

If you get it working I would happy bring it back into the code base as a separate version? 

I know OKHttp has an experimental eventsource library so I could probably build one of those for you, but it would need to go into the backlog.

**Reply by pontamilselvan** [reply]
got another conflict :(

May 27, 2021 9:10:36 AM com.sun.xml.ws.server.sei.TieHandler createResponse
SEVERE: javax.ws.rs.core.UriBuilder.uri(Ljava/lang/String;)Ljavax/ws/rs/core/UriBuilder;
java.lang.AbstractMethodError: javax.ws.rs.core.UriBuilder.uri(Ljava/lang/String;)Ljavax/ws/rs/core/UriBuilder;
        at javax.ws.rs.core.UriBuilder.fromUri(UriBuilder.java:96)
        at org.glassfish.jersey.client.JerseyWebTarget.<init>(JerseyWebTarget.java:48)
        at org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:320)
        at org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:56)
        at io.featurehub.client.jersey.JerseyClient.makeEventSourceTarget(JerseyClient.java:78)
        at io.featurehub.client.jersey.JerseyClient.<init>(JerseyClient.java:54)
        at io.featurehub.client.jersey.JerseyClient.<init>(JerseyClient.java:43)


i am seeing UriBuilder in below two jars

jersey-common-2.28.jar
jakarta.ws.rs-api-2.1.5.jar

**Comment by rvowles** [comment]
Heya - which approach are you using? Did you fork the Jersey SDK? Its
probably better if we tackle this in a proper Github repo for you - get a
Jersey 1 compliant SDK?

On Fri, May 28, 2021 at 1:24 AM pontamilselvan ***@***.***>
wrote:

> got another conflict :(
>
> May 27, 2021 9:10:36 AM com.sun.xml.ws.server.sei.TieHandler createResponse
> SEVERE:
> javax.ws.rs.core.UriBuilder.uri(Ljava/lang/String;)Ljavax/ws/rs/core/UriBuilder;
> java.lang.AbstractMethodError:
> javax.ws.rs.core.UriBuilder.uri(Ljava/lang/String;)Ljavax/ws/rs/core/UriBuilder;
> at javax.ws.rs.core.UriBuilder.fromUri(UriBuilder.java:96)
> at org.glassfish.jersey.client.JerseyWebTarget.(JerseyWebTarget.java:48)
> at org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:320)
> at org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:56)
> at
> io.featurehub.client.jersey.JerseyClient.makeEventSourceTarget(JerseyClient.java:78)
> at io.featurehub.client.jersey.JerseyClient.(JerseyClient.java:54)
> at io.featurehub.client.jersey.JerseyClient.(JerseyClient.java:43)
>
> i am seeing UriBuilder in below tow jars
>
> jersey-common-2.28.jar
> jakarta.ws.rs-api-2.1.5.jar
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/featurehub-io/featurehub/discussions/410#discussioncomment-792969>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAANTWHKA75FDO35VNQEBXDTPZBZ7ANCNFSM45S7VP3Q>
> .
>


-- 
---
Richard Vowles,
Full stack - from Kubernetes, through Node & Java, Web and Mobile
development in Flutter - software developer for hire!

ph: +64275467747

**Comment by rvowles** [comment]
I'm sorry to say that Jersey 1 does not actually support Server Sent Events, so I can't build you a client using Jersey 1 that supports it. I will need to enhance the OKHttp one to support it for you instead.

**Reply by pontamilselvan** [reply]
then i need to upgrade to jersey 2 is a another option

**Reply by rvowles** [reply]
I'm a fair bit through supporting one for OKHttp, which is just an HTTP Client library that would allow you to stay with Jersey 1. Jersey 1 is pretty old though, but I don't want our library to force you into a whole lot of work.

**Comment by rvowles** [comment]
https://github.com/featurehub-io/featurehub/issues/416

**Comment by rvowles** [comment]
Hi @pontamilselvan I presume you no longer require this?

**Reply by pontamilselvan** [reply]
Hi, I upgraded my code from jersey 1.x to jersey 2.3.

**Reply by rvowles** [reply]
Ok, will mark this answered :-)
